src/main/java/org/traccar/handler/FilterHandler.java

```diff
@@ -54,22 +58,23 @@ public FilterHandler(Config config) {
         filterDistance = config.getInteger(Keys.FILTER_DISTANCE);
         filterMaxSpeed = config.getInteger(Keys.FILTER_MAX_SPEED);
         filterMinPeriod = config.getInteger(Keys.FILTER_MIN_PERIOD) * 1000;
+        filterRelative = config.getBoolean(Keys.FILTER_RELATIVE);
         skipLimit = config.getLong(Keys.FILTER_SKIP_LIMIT) * 1000;
         skipAttributes = config.getBoolean(Keys.FILTER_SKIP_ATTRIBUTES_ENABLE);
     }
 
     private boolean filterInvalid(Position position) {
         return filterInvalid && (!position.getValid()
-           || position.getLatitude() > 90 || position.getLongitude() > 180
-           || position.getLatitude() < -90 || position.getLongitude() < -180);
+                || position.getLatitude() > 90 || position.getLongitude() > 180
+                || position.getLatitude() < -90 || position.getLongitude() < -180);
     }
 
     private boolean filterZero(Position position) {
         return filterZero && position.getLatitude() == 0.0 && position.getLongitude() == 0.0;
     }
 
     private boolean filterDuplicate(Position position, Position last) {
-        if (filterDuplicate && last != null && position.getFixTime().equals(last.getFixTime())) {
+        if (filterDuplicate && last != null && position.getFixTime().getTime() == last.getFixTime().getTime()) {
```

Comment:
Why is this needed?

Reply:
If we're going to store milli/micro/nanoseconds in the TIMESTAMP field in the DB, then it means we would in fact compare float numbers. So maybe to filter out duplicates we should compare `getTime()` with a specified precision.

Reply:
Sorry, where did you get floating numbers and nanoseconds? As far as I can tell, `equals` does exactly the same thing as you do manually:

https://docs.oracle.com/javase/8/docs/api/java/util/Date.html#equals-java.lang.Object-

Reply:
Yes, currently it does the same thing. But, if we make this change https://github.com/traccar/traccar/pull/4746, then database would store time with milliseconds/nanosecond precision (as float). Hence, comparing new position (converted from text) and preceding (from DB) might not be equal in theory. That is why I suggest to do comparison with a certain precision. I can do additional commit for that. Do you agree?

Reply:
I thought you planned to fix database accuracy?

Reply:
Let's leave it as it is now. And deal with it in the separate PR.

Reply:
Agree but this change should be reverted because it doesn't really do anything.

Reply:
Done.